packages/webrtc-core/src/media/track.ts

```diff
@@ -0,0 +1,198 @@
+/* eslint-disable no-underscore-dangle */
+import { EventMap } from 'typed-emitter';
+import { EventEmitter } from '../event-emitter';
+import { logger } from '../util/logger';
+import { MediaStreamTrackKind } from '../peer-connection';
+
+export enum Events {
+  Ended = 'ended',
+  Muted = 'muted',
+  /**
+   * Fires when the published state of a LocalTrack changes.  A value of 'true' indicates the
+   * track has been published, and 'false' indicates that it is no longer published.  Tracks are
+   * unpublished by default, and must be published explicitly (no event will be fired to indicate
+   * the initial state of 'unpublished').
+   */
+  PublishedStateUpdate = 'published-state-update',
+  /**
+   * Fires when there has been a change in the underlying track.
+   */
+  UnderlyingTrackChange = 'underlying-track-change',
+}
+
+export interface TrackState {
+  id: string;
+  label: string;
+  muted: boolean;
+}
+
+export type TrackEndEvent = {
+  trackState: TrackState;
+};
+
+export type TrackMuteEvent = {
+  trackState: TrackState;
+};
+// eslint-disable-next-line no-shadow
+export enum TrackStatus {
+  ENDED = 'ended',
+  LIVE = 'live',
+}
+
+// eslint-disable-next-line no-shadow
+export enum TrackKind {
+  AUDIO = 'audio',
+  VIDEO = 'video',
+}
+
+export type TrackPublishEvent = {
```

Comment:
this is already defined in local-track.ts

Reply:
done